src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,103 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+
+type Props = {
+  items: string[],
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<Props> = ({
+  items,
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesCount = Math.ceil(total / perPage);
+  const pages = [];
+  const isFirstPage = currentPage === 1;
+  const isLastPage = currentPage === pagesCount;
+
+  for (let i = 1; i <= pagesCount; i += 1) {
+    pages.push(i);
+  }
```

Comment:
it will be executed every time when the state change, so make sense to move it to the useEffect